### PR TITLE
Add inlineScriptAsset to inline built asset contents by entry path

### PR DESCRIPTION
**Pull request by cyruscollier**
## Summary

Adds `ModuleAssets::inlineScriptAsset()`, an inlining variant of `inlineScript` that takes just the relative script entry path (like `enqueueScript`) instead of a raw content string.

- **Production:** resolves the entry through the Vite manifest to the hashed build output and inlines its file contents.
- **Dev server:** emits `<script type="module">import "<devUrl>";</script>` so Vite's on-the-fly transformed module (with its imports and HMR) runs correctly, rather than fetching and inlining transformed source (which would break relative imports).

### Optional handle / top-of-head

`inlineScript` and `inlineScriptAsset` now take the content/src **first** and an **optional** `$handle`:

- With a handle: existing behavior — inlined onto the enqueued script via `wp_add_inline_script` (prod) or the dev echo path.
- Without a handle: `wp_add_inline_script` is bypassed and the script is printed directly at `wp_head` priority 1 — **ahead of the enqueued scripts** (priority 9) — so callers can emit an inline asset at the top of the head without tying it to an enqueued dependency.

`inlineScriptData` keeps its public signature; only its internal call is reordered. The `UIFramework` call sites are updated to the new argument order (the only in-repo direct callers of `inlineScript`).

Supporting changes:
- `inlineScript` gains a `module` flag to emit `type="module"` on the dev-server output.
- New private helpers `assetContents()` (reads the built file) and `scriptPath()` (mirrors the existing `stylePath()`).
- Minor refactor extracting `assetPathRelative()` out of `assetUrlRelative()`.

> **Note:** flipping the `inlineScript`/`inlineScriptAsset` argument order is a breaking change for any out-of-repo consumers calling those methods directly (`inlineScriptData` is unaffected).

## Tests

- `test_inlineScriptAsset_prod` — built file's contents inlined on the registered handle.
- `test_inlineScriptAsset_prod_withoutHandle_printsInHead` — handle-less call prints the script on `wp_head`.
- `test_inlineScriptAsset_dev` — `type="module"` import tag renders on `wp_head`.
- Adds the `dist/assets/test-abcde.js` build-output fixture the manifest already maps `scripts/test.js` to.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/Framework/ModuleAssets.php [hunk 1]
```diff
@@ -226,15 +249,30 @@ private function assetUrl(FilePath $assetPath): string
...
+    }
+
+    private function assetContents(FilePath $assetPath): string
+    {
+        $buildAssetPath = $this->buildAssetPath($assetPath);
+        if (!$buildAssetPath) {
+            Logger::warning('Production build path not found for asset: ' . $assetPath->value());
+            return '';
+        }
+        return (string) file_get_contents($buildAssetPath->value());
```
Comment (jeremystrom) [inline src/Framework/ModuleAssets.php]: **Question:** What is the failure contract when a built asset can't be read?

Resolution failures (missing manifest file or missing manifest key) already log-and-skip via `Logger::warning` in `assetContents()`. The one truly silent case is a file that *is* in the manifest but unreadable on disk: `file_get_contents()` returns `false`, the cast turns it into `''`, and `inlineScriptAsset()` bails at line 171 with no log. A legitimately empty built file is indistinguishable from that read failure — both leave required JavaScript off the page with no signal.

```php
private function assetContents(FilePath $assetPath): string
{
    $buildAssetPath = $this->buildAssetPath($assetPath);
    if (!$buildAssetPath) {
        Logger::warning('Production build path not found for asset: ' . $assetPath->value());
        return '';
    }
    return (string) file_get_contents($buildAssetPath->value());
}
```

The policy question: should production hard-fail, log-and-skip (the current behavior for resolution failures), or stay silent — and should "empty build output" be distinguishable from "read failed"? Two minor related notes: the dev-path guard `if (!$url) return;` at lines 163–165 is dead code (dev-mode `assetUrl()` always returns a non-empty URL), and a `wp_json_encode()` failure at line 166 would emit the invalid statement `import ;` (only reachable with invalid UTF-8 in the URL — near-impossible in practice).

**Verify:** Whatever policy is chosen, at minimum mirror the line-263 warning when `file_get_contents()` returns `false`, and add tests for missing-manifest-entry, unreadable-file, and empty-file cases — none exist today.

File: src/Framework/ModuleAssets.php [hunk 2]
```diff
@@ -133,22 +133,45 @@ public function enqueueBlockStyle(string $blockName, string $src): void
...
+            $content = sprintf('import %s;', wp_json_encode($url, JSON_UNESCAPED_SLASHES));
+            $this->inlineScript($content, $handle, $position, true);
+            return;
+        }
+        $content = $this->assetContents($assetPath);
+        if (!$content) {
+            return;
         }
+        $this->inlineScript($content, $handle, $position);
     }
```
Comment (jeremystrom) [inline src/Framework/ModuleAssets.php:193]: **Question:** Is the dev/prod semantics divergence in `inlineScriptAsset()` acceptable?

The dev branch emits `<script type="module">import "…";</script>` (module scope, implicit strict mode, deferred execution) while the prod branch inlines the raw built bytes as a classic script (global scope, synchronous). This divergence is largely inherent to Vite — the dev server serves unbundled ES modules that must be `import`ed and cannot be inlined raw — and `inlineScriptAsset()` currently has zero production callers, so nothing is broken today.

The risk to confirm: if a built entry ever emits ESM format (top-level `import`/`export`), production would parse-fail in a way dev testing can never surface, since dev always uses the module wrapper. Additionally, when a handle is passed in dev mode, the script attaches to a generic hook at priority 10 rather than `wp_add_inline_script()`'s handle-adjacent placement — the "immediately before this specific script" guarantee is lost while the dev server is active.

**Verify:** Confirm the Vite build outputs self-contained (IIFE/UMD) bundles for entries intended for inlining, and state whether any planned caller depends on module semantics or handle-relative ordering. If the divergence is intentional, a docblock note on `inlineScriptAsset()` would prevent this question from recurring.

File: src/Framework/ModuleAssets.php (hunk 1 above)
Comment (jeremystrom) [inline src/Framework/ModuleAssets.php]: **Question:** Is unbounded, verbatim inlining the intended usage?

Built asset contents are re-read from disk on every call (only the manifest is cached), and the complete file bytes — including any `sourceMappingURL` directive, license header, or debug comments — are embedded verbatim into every HTML response with no size or content guard. None of this is a defect: inlining defeats HTTP caching by design, Vite prod builds emit external `.map` files (the trailing comment is harmless), and license headers often must be preserved. But nothing in the API constrains it to small critical scripts, and the trade-off is undocumented.

**Verify:** If the feature is intended only for small critical scripts (e.g. the no-JS class swap), say so in the docblock; consider a size guard as cheap insurance against future misuse with large bundles.

File: src/Framework/ModuleAssets.php [hunk 3]
```diff
@@ -133,22 +133,45 @@ public function enqueueBlockStyle(string $blockName, string $src): void
...
             wp_add_inline_script($handle, $content, $position);
             return;
         }
         $isHeader = $position !== 'after';
         $hook = $isHeader ? (is_admin() ? 'admin_head' : 'wp_head') : (is_admin() ? 'admin_footer' : 'wp_footer');
-        $callback = function () use ($content) {
-            echo "<script>{$content}</script>";
+        $type = $module ? ' type="module"' : '';
+        $callback = function () use ($content, $type) {
+            echo "<script{$type}>{$content}</script>";
```
Comment (jeremystrom) [inline src/Framework/ModuleAssets.php]: **Suggestion:** Use `wp_get_inline_script_tag()` instead of the hand-built tag

The handle-less path echoes raw, unescaped content: a literal `</script>` sequence in a built file would terminate the HTML script element early and break the document. The content is the site's own built JavaScript, so this is robustness rather than an XSS vector — but the fix is one line and also resolves the framework-abstraction bypass (manual tag/attribute interpolation instead of WordPress' inline-script functions).

<details>
<summary>Current code</summary>

```php
$type = $module ? ' type="module"' : '';
$callback = function () use ($content, $type) {
    echo "<script{$type}>{$content}</script>";
};
```

</details>

```php
$attributes = $module ? ['type' => 'module'] : [];
$callback = function () use ($content, $attributes) {
    echo wp_get_inline_script_tag($content, $attributes);
};
```

**Verify:** This WordPress version (7.0.1) escapes via `WP_HTML_Tag_Processor::set_modifiable_text()` — its docblock example shows `console.log( "</script>" )` being safely escaped. The existing rendered-output tests (`test_inlineScriptAsset_prod_withoutHandle_printsInHead`, `test_inlineScriptAsset_dev`) will catch any tag-shape change; run the suite.

File: src/Framework/ModuleAssets.php [hunk 4]
```diff
@@ -133,22 +133,45 @@ public function enqueueBlockStyle(string $blockName, string $src): void
...
         };
+        // Without a handle the script isn't tied to an enqueued dependency, so print it at the
+        // very top of the head, ahead of the enqueued scripts (which output at priority 9).
+        $priority = $handle ? 10 : 1;
         if (did_action($hook)) {
             $callback();
         } else {
-            add_action($hook, $callback);
+            add_action($hook, $callback, $priority);
+        }
```
Comment (jeremystrom) [inline src/Framework/ModuleAssets.php]: **Suggestion:** The declared priorities are dead on the standard call path

WordPress increments the action counter at the top of `do_action()`, and core fires `wp_enqueue_scripts` nested inside `wp_head` at priority 1. So any call arriving via `Module::enqueueFrontendAssets()` sees `did_action('wp_head')` already truthy and always takes the immediate-echo branch — `$priority = $handle ? 10 : 1` never applies on that path, and head output lands ahead of enqueued scripts only by luck of the nesting. The handle-less `position: 'after'` branch lands on `wp_footer` at priority 1, which runs *before* `wp_print_footer_scripts` (priority 20) — inverting the conventional meaning of "after" — and the admin branches don't match the priority-9 premise in the comment either.

```php
// Without a handle the script isn't tied to an enqueued dependency, so print it at the
// very top of the head, ahead of the enqueued scripts (which output at priority 9).
$priority = $handle ? 10 : 1;
if (did_action($hook)) {
    $callback();
} else {
    add_action($hook, $callback, $priority);
}
```

Triage context that lowers the urgency: the only real callers today pass handles in production, which routes to `wp_add_inline_script()` before this code is reached, so the broken branches are currently unexercised. Also note `Hooks::callOrAddAction()` is **not** a drop-in replacement despite the apparent duplication — it always registers the action *and* fires the callback immediately when `did_action()` is true (double-fire semantics), unlike this mutually-exclusive if/else.

**Verify:** Any redesign should add the missing ordering test: register a competing `wp_head` handler that prints enqueued-script output at priority 9 and assert the inline script precedes it. The current test passes identically at priority 1, 10, or none.

File: src/Framework/ModuleAssets.php [hunk 5]
```diff
@@ -133,22 +133,45 @@ public function enqueueBlockStyle(string $blockName, string $src): void
         wp_enqueue_block_style($blockName, $args);
     }
 
-    public function inlineScript(string $handle, string $content, $position = null): void
+    public function inlineScript(string $content, string $handle = '', $position = null, bool $module = false): void
```
Comment (jeremystrom) [inline src/Framework/ModuleAssets.php:148]: **Suggestion:** The argument flip is silently type-compatible with the old signature

`inlineScript()` changed from `(string $handle, string $content, …)` to `(string $content, string $handle = '', …)`. Both parameters are plain strings, so an unmigrated old-order call type-checks and silently swaps semantics: `wp_add_inline_script()` on the JavaScript-as-handle returns `false`, that return is discarded, and in dev mode the old handle string prints as script content. Triage confirmed every reachable consumer is migrated (both `UIFramework.php` call sites; the sibling `sitchco-parent-theme` and `roundabout` themes only use `inlineScriptData()`, which kept handle-first order) — so the exposure is limited to unmigrated external consumers and future contributors. The class now also mixes conventions: `registerScript`/`enqueueScript`/`inlineScriptData` are handle-first while `inlineScript`/`inlineScriptAsset` are content/src-first.

**Verify:** If any consumers live outside the checked repos, grep them for `inlineScript(`. Options if hardening is wanted: a cheap guard (warn when `$handle` contains whitespace/semicolons, or when `wp_add_inline_script()` returns false), or a distinct method name so old-order calls fail loudly.

File: src/Framework/ModuleAssets.php [hunk 6]
```diff
@@ -226,15 +249,30 @@ private function assetUrl(FilePath $assetPath): string
         return '';
     }
 
+    private function assetPathRelative(string $relativePath): FilePath
+    {
+        return str_starts_with($relativePath, $this->moduleAssetsPath->value())
+            ? new FilePath($this->moduleAssetsPath->value())
+            : $this->moduleAssetsPath->append($relativePath);
```
Comment (jeremystrom) [inline src/Framework/ModuleAssets.php]: **Suggestion:** Delete (or fix) the dead absolute-path branch in `assetPathRelative()`

The `str_starts_with` branch is provably unreachable — every caller prepends a relative subdirectory (`scripts/`, `styles/`, `images/`), which can never start with the absolute assets path — and it is also wrong for the one case it tests: it returns the assets *root*, discarding the requested file entirely. Git history confirms this PR extracted the block verbatim from `assetUrlRelative()`, so the behavior is relocated, not introduced; but since the PR touched exactly these lines, cleaning it up is in scope. The new `scriptPath()`/`stylePath()` helpers already do a plain append with no special case.

<details>
<summary>Current code</summary>

```php
private function assetPathRelative(string $relativePath): FilePath
{
    return str_starts_with($relativePath, $this->moduleAssetsPath->value())
        ? new FilePath($this->moduleAssetsPath->value())
        : $this->moduleAssetsPath->append($relativePath);
}
```

</details>

**Verify:** `rg 'assetPathRelative|assetUrlRelative'` shows all inputs are `"scripts/…"`, `"styles/…"`, `"images/…"` literals. Deleting the ternary in favor of the plain append should pass the full suite unchanged.

File: src/Framework/ModuleAssets.php [hunk 7]
```diff
@@ -226,15 +249,30 @@ private function assetUrl(FilePath $assetPath): string
...
+
+    private function assetContents(FilePath $assetPath): string
+    {
+        $buildAssetPath = $this->buildAssetPath($assetPath);
+        if (!$buildAssetPath) {
+            Logger::warning('Production build path not found for asset: ' . $assetPath->value());
+            return '';
+        }
+        return (string) file_get_contents($buildAssetPath->value());
+    }
```
Comment (jeremystrom) [inline src/Framework/ModuleAssets.php]: **Suggestion:** Extract the duplicated manifest-lookup-and-warn logic

`assetUrl()` and `assetContents()` each independently call `buildAssetPath()` and emit the byte-identical warning `'Production build path not found for asset: …'`. Beyond the drift risk, `buildAssetPath()` returns `null` for both a missing manifest file and a missing manifest key, so neither call site can distinguish the two. A single shared resolve-and-warn helper is the natural home for whatever failure signaling comes out of the failure-policy question raised on line 266, including separating "manifest missing" from "key missing."

**Verify:** Pure refactor with no behavioral change; the existing prod-path tests cover both call sites.

File: tests/Framework/ModuleAssetsTest.php [hunk 8]
```diff
@@ -255,6 +255,40 @@ public function test_inlineScript_dev()
...
+        $this->devAssets->inlineScriptAsset('test.js', $handle);
+        ob_start();
+        do_action('wp_head');
+        $html_out = ob_get_clean();
+        $this->assertStringContainsString(
+            '<script type="module">import "https://example.org:5173/Fakes/ModuleTester/assets/scripts/test.js";</script>',
+            $html_out,
+        );
+        $this->assertViteClientEnqueued();
+    }
```
Comment (jeremystrom) [inline tests/Framework/ModuleAssetsTest.php:302]: **Suggestion:** Cover the four untested branches

All four gaps were verified against the test file:

1. **Dev server + no handle** — the combination that makes the handle-less feature useful in dev; `test_inlineScriptAsset_dev` always passes a handle.
2. **`position: 'after'`** — no new test passes a position, so `$isHeader` is never false and the `wp_footer` branch never runs.
3. **Admin hooks** — `admin_head`/`admin_footer` selection is unreached (and the constructor forces `isDevServer = false` under `is_admin()`).
4. **The `did_action()` immediate-echo branch** — every test calls `inlineScriptAsset()` *before* `do_action('wp_head')`, so only the `add_action` branch executes. This is the branch real production traffic actually hits (calls arrive nested inside `wp_head`), meaning the suite tests the branch requests rarely take and skips the one they always take.

**Verify:** Prioritize the immediate-echo and dev-no-handle cases. If the scheduling is redesigned per the priority suggestion on `ModuleAssets.php:148-155`, write these against the new behavior rather than pinning the current one.

File: tests/Framework/ModuleAssetsTest.php [hunk 9]
```diff
@@ -255,6 +255,40 @@ public function test_inlineScript_dev()
...
+
+    public function test_inlineScriptAsset_dev()
+    {
+        $handle = ModuleTester::hookName('test');
+        $this->devAssets->enqueueScript($handle, 'test.js');
+        $this->devAssets->inlineScriptAsset('test.js', $handle);
+        ob_start();
+        do_action('wp_head');
+        $html_out = ob_get_clean();
+        $this->assertStringContainsString(
```
Comment (jeremystrom) [inline tests/Framework/ModuleAssetsTest.php:297]: **Suggestion:** Tighten the three new tests

Three mechanical fixes:

- **Global-state leak:** only `test_inlineScriptAsset_prod` calls `resetWPDependencies()`; `test_inlineScriptAsset_dev` mutates `$GLOBALS['wp_scripts']`/`$GLOBALS['wp_script_modules']` without it, so the `restoreWPDependencies()` tearDown is a no-op and registrations leak into subsequent tests. (Triage note: this is a pre-existing convention gap — the older `test_inlineScript_prod`/`test_inlineScript_dev` also skip the reset — but it's the one item with real cross-test blast radius if fixtures or ordering change.)
- **Unterminated prefix assertion** at line 274: `"<script>console.log('built test asset');"` has no closing tag, so a missing `</script>` or appended garbage passes undetected. Siblings in the same file use full-tag assertions (`assertHTMLEquals` at lines 164/171/214) — use that surface.
- **Internal-storage coupling** at line 265: asserting on `trim($registered->extra['before'][1])` couples to WordPress internals and the `trim()` masks the fixture's trailing newline; the rendered-HTML assertion style used elsewhere is more robust.

**Verify:** Add `resetWPDependencies()` to the dev test, swap the two assertions to full rendered tags, and run the suite.

File: src/Framework/ModuleAssets.php (hunk 2 above)
Comment (cyruscollier) [reply src/Framework/ModuleAssets.php:193]: Yes, the Vite production build always emits IIFE entry files, and the differing output format and timing behavior with the dev server is unavoidable, but acceptable.

File: src/Framework/ModuleAssets.php (hunk 5 above)
Comment (cyruscollier) [reply src/Framework/ModuleAssets.php:148]: I verified the only caller of this is inside the same repo and has been updated with this PR.

File: src/Framework/ModuleAssets.php (hunk 3 above)
Comment (cyruscollier) [reply src/Framework/ModuleAssets.php]: Done in 7baf7d54dea929250fa4eb0ddd90c4af5a6e1dc8 — `inlineScript()` now echoes via `wp_get_inline_script_tag()` (drops the hand-built tag and the `$type` string). Existing rendered-output tests updated for the newline-wrapped shape and pass.

File: src/Framework/ModuleAssets.php (hunk 4 above)
Comment (cyruscollier) [reply src/Framework/ModuleAssets.php]: Addressed in 7baf7d54dea929250fa4eb0ddd90c4af5a6e1dc8 (targeted cleanup, structure kept). Handle-less head priority corrected `1 -> 0` so it lands ahead of enqueued scripts (priority 9); the misleading comment is replaced with a docblock on `inlineScript()` covering hook selection, the priority choice, and that the prod+handle path is handled earlier by `wp_add_inline_script()`. Left `Hooks::callOrAddAction()` alone per your note about its double-fire semantics. Added `test_inlineScriptAsset_withoutHandle_printsBeforePriority9HeadOutput`, which stands up a competing priority-9 handler and asserts the inline script precedes it.

File: src/Framework/ModuleAssets.php (hunk 1 above)
Comment (cyruscollier) [reply src/Framework/ModuleAssets.php]: Resolved in 7baf7d54dea929250fa4eb0ddd90c4af5a6e1dc8 with the log-and-skip policy (consistent with the rest of the class — no hard-fail). `assetContents()` now logs `Failed to read built asset: …` when `file_get_contents()` returns `false`, so a read failure is no longer silent. The manifest-resolution warning is shared via the new `resolveBuildAssetPath()` helper (see the @267 thread). Added tests cover the empty/unreadable and missing-entry cases via the branch coverage. (Left the near-impossible `wp_json_encode`-of-URL edge alone.)

File: src/Framework/ModuleAssets.php (hunk 7 above)
Comment (cyruscollier) [reply src/Framework/ModuleAssets.php]: Done in 7baf7d54dea929250fa4eb0ddd90c4af5a6e1dc8 — extracted `resolveBuildAssetPath()`, now the single resolve-and-warn home used by both `assetUrl()` and `assetContents()`, eliminating the duplicated warning.

File: src/Framework/ModuleAssets.php (hunk 6 above)
Comment (cyruscollier) [reply src/Framework/ModuleAssets.php]: Done in 7baf7d54dea929250fa4eb0ddd90c4af5a6e1dc8 — `assetPathRelative()` reduced to the plain `->append()`; the unreachable/incorrect `str_starts_with` ternary is gone. Full suite passes unchanged.

File: src/Framework/ModuleAssets.php (hunk 1 above)
Comment (cyruscollier) [reply src/Framework/ModuleAssets.php]: Addressed in 7baf7d54dea929250fa4eb0ddd90c4af5a6e1dc8 with a docblock (no runtime size guard, by decision). `inlineScriptAsset()` now documents that it embeds the full built bytes verbatim on every response and is intended for small, critical scripts since inlining defeats HTTP caching by design.

File: tests/Framework/ModuleAssetsTest.php (hunk 9 above)
Comment (cyruscollier) [reply tests/Framework/ModuleAssetsTest.php:297]: Done in 7baf7d54dea929250fa4eb0ddd90c4af5a6e1dc8. `resetWPDependencies()` added to the dev tests (tearDown restore is no longer a no-op); the unterminated-prefix assertion is now a full rendered-tag check; and the prod test asserts against rendered `do_item()` output instead of `trim($registered->extra['before'][1])`, so it no longer couples to WP internals or masks the trailing newline.

File: tests/Framework/ModuleAssetsTest.php (hunk 8 above)
Comment (cyruscollier) [reply tests/Framework/ModuleAssetsTest.php:302]: Done in 7baf7d54dea929250fa4eb0ddd90c4af5a6e1dc8 — added tests for all four: immediate-echo (call after `do_action('wp_head')`), dev + no handle, `position: 'after'` (footer), and admin (`admin_head`), plus the priority-ordering test from the @155 thread. Note: because `resetWPDependencies()` wipes the script registry, each echo test `remove_all_actions()`s the hook it fires so WP's default script printing doesn't warn about the (unrelated) unregistered deps in the test install. Full suite is green.
